x-pack/plugins/ml/server/lib/alerts/register_jobs_monitoring_rule_type.ts

```diff
@@ -0,0 +1,108 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { KibanaRequest } from 'kibana/server';
+import { ML_ALERT_TYPES } from '../../../common/constants/alerts';
+import { PLUGIN_ID } from '../../../common/constants/app';
+import { MINIMUM_FULL_LICENSE } from '../../../common/license';
+import {
+  anomalyDetectionJobsHealthRuleParams,
+  AnomalyDetectionJobsHealthRuleParams,
+} from '../../routes/schemas/alerting_schema';
+import { RegisterAlertParams } from './register_ml_alerts';
+import {
+  ActionGroup,
+  AlertInstanceContext,
+  AlertInstanceState,
+  AlertTypeState,
+} from '../../../../alerting/common';
+
+export type AnomalyDetectionJobsHealthAlertContext = {
+  jobIds: string[];
+  message: string;
+} & AlertInstanceContext;
+
+export const ANOMALY_DETECTION_JOB_REALTIME_ISSUE = 'anomaly_detection_realtime_issue';
+
+export type AnomalyDetectionJobRealtimeIssue = typeof ANOMALY_DETECTION_JOB_REALTIME_ISSUE;
+
+export const REALTIME_ISSUE_DETECTED: ActionGroup<AnomalyDetectionJobRealtimeIssue> = {
+  id: ANOMALY_DETECTION_JOB_REALTIME_ISSUE,
+  name: i18n.translate('xpack.ml.jobsHealthAlertingRule.actionGroupName', {
+    defaultMessage: 'Realtime issue detected',
+  }),
+};
+
+export function registerJobsMonitoringRuleType({
+  alerting,
+  mlServicesProviders,
+  logger,
+}: RegisterAlertParams) {
+  alerting.registerType<
+    AnomalyDetectionJobsHealthRuleParams,
+    AlertTypeState,
+    AlertInstanceState,
+    AnomalyDetectionJobsHealthAlertContext,
+    AnomalyDetectionJobRealtimeIssue
+  >({
+    id: ML_ALERT_TYPES.AD_JOBS_HEALTH,
+    name: i18n.translate('xpack.ml.jobsHealthAlertingRule.name', {
+      defaultMessage: 'Jobs health monitoring',
+    }),
+    actionGroups: [REALTIME_ISSUE_DETECTED],
+    defaultActionGroupId: ANOMALY_DETECTION_JOB_REALTIME_ISSUE,
+    validate: {
+      params: anomalyDetectionJobsHealthRuleParams,
+    },
+    actionVariables: {
+      context: [
+        {
+          name: 'jobIds',
+          description: i18n.translate(
+            'xpack.ml.alertTypes.jobsHealthAlertingRule.alertContext.jobIdsDescription',
+            {
+              defaultMessage: 'List of job IDs that triggered the alert instance',
+            }
+          ),
+        },
+        {
+          name: 'message',
+          description: i18n.translate(
+            'xpack.ml.alertTypes.jobsHealthAlertingRule.alertContext.messageDescription',
+            {
+              defaultMessage: 'Alert info message',
+            }
+          ),
+        },
+      ],
+    },
+    producer: PLUGIN_ID,
+    minimumLicenseRequired: MINIMUM_FULL_LICENSE,
+    isExportable: true,
+    async executor({ services, params, alertId, state, previousStartedAt, startedAt }) {
+      const fakeRequest = {} as KibanaRequest;
+      const { getTestsResults } = mlServicesProviders.jobsHealthServiceProvider(
```

Comment:
Any errors thrown by this call will be bubbled up to the alerting task runner and logged as an error but if you want any more detailed information or to log debug messages, suggest wrapping in your own try/catch before throwing the error.